src/pages/main.tsx

```diff
@@ -7,75 +7,56 @@ import Modal, { OVERLAY_ID, CLOSE_ID } from 'components/modal';
 import ItemCard from 'components/itemCard';
 import DownloadIndicator from 'components/downloadIndicator';
 
-interface IMainPageState {
-  itemsData: IItemData[];
-  isModalVisible: boolean;
-  itemModalIdx: number;
-  errMsg: string;
-  isDataLoading: boolean;
-}
-
-class Main extends React.Component<unknown, IMainPageState> {
-  constructor(props = {}) {
-    super(props);
-    this.state = {
-      itemsData: [],
-      isModalVisible: false,
-      itemModalIdx: 0,
-      errMsg: '',
-      isDataLoading: true,
-    };
-  }
-
-  componentDidMount = () => {
-    this.getItemsData('');
-  };
+function Main() {
+  const [itemsData, setItemsData] = useState([] as IItemData[]);
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const itemModalIdx = useRef(0);
+  const errMsg = useRef('');
+  const [isDataLoading, setIsDataLoading] = useState(true);
 
-  getItemsData = async (search: string) => {
-    this.setState({ itemsData: [], isDataLoading: true });
+  const getItemsData = async (search: string) => {
+    setItemsData([]);
+    setIsDataLoading(true);
     const data = await getItems(search);
-    this.setState({ itemsData: data.items, errMsg: data.errMsg, isDataLoading: false });
+    errMsg.current = data.errMsg;
+    setIsDataLoading(false);
+    setItemsData(data.items);
   };
 
-  showModal: React.MouseEventHandler = (e) => {
-    this.setState({ isModalVisible: true, itemModalIdx: parseInt(e.currentTarget.id) });
+  const showModal: React.MouseEventHandler = (e) => {
+    itemModalIdx.current = parseInt(e.currentTarget.id);
```

Comment:
Again, why not using state for it? You can save id of item you want to show in the modal